### PR TITLE
[Naitei][Ecommerce] [Admin]Delete category

### PR DESCRIPTION
**Pull request by SuKay712**
## Related Tickets
https://edu-redmine.sun-asterisk.vn/issues/78413

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
Screenshot from 2024-07-28 12-23-47
Screenshot from 2024-07-28 12-22-03

Screenshot from 2024-07-30 13-47-40

### CODE REVIEW

**Comment by SuKay712** [comment]
ready

File: app/views/categories/_category.html.erb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<tr class="table-row">
+  <td class="table-data"><%= category.id %></td>
```
Comment (ghost) [inline app/views/categories/_category.html.erb]: Sao không dùng index để đánh số thứ tự em ơi
Em tham khảo cách lấy số thứ tự khi dùng partial nhé

**Review by ghost — COMMENTED** [review]
cmt

File: app/views/categories/_category.html.erb (hunk 1 above)
Comment (SuKay712) [reply app/views/categories/_category.html.erb]: dạ

**Comment by SuKay712** [comment]
ready

File: app/views/admin/categories/index.html.erb [hunk 2]
```diff
@@ -3,14 +3,17 @@
...
         <th class="table-header"><%= t "category.name" %></th>
         <th class="table-header"><%= t "create_time" %></th>
         <th class="table-header"><%= t "update_time" %></th>
         <th class="table-header"><%= t "action" %></th>
       </tr>
     </thead>
     <tbody id="categories_list">
-      <%= render @categories %>
+      <% @categories.each_with_index do |category, index| %>
+        <%= render partial: "category", locals: { category: category, index: index + 1 } %>
```
Comment (ThaiKhacDuoc) [inline app/views/admin/categories/index.html.erb]: Đưa vào trong Setting huynh ơi
`1 `

**Review by ThaiKhacDuoc — COMMENTED** [review]
cmt

File: app/views/admin/categories/index.html.erb (hunk 2 above)
Comment (SuKay712) [reply app/views/admin/categories/index.html.erb]: oke

File: app/views/admin/categories/index.html.erb [hunk 3]
```diff
@@ -3,14 +3,17 @@
...
         <th class="table-header"><%= t "create_time" %></th>
         <th class="table-header"><%= t "update_time" %></th>
         <th class="table-header"><%= t "action" %></th>
       </tr>
     </thead>
     <tbody id="categories_list">
-      <%= render @categories %>
+      <% @categories.each_with_index do |category, index| %>
+        <%= render partial: "category", locals: { category: category, index: index + Settings.index_range } %>
+      <% end %>
```
Comment (ghost) [inline app/views/admin/categories/index.html.erb]:
```suggestion
      <%= render partial: "category", collection: @categories, as: :category %>
```
Bên trong muốn lấy index thì dùng category_counter
https://coderwall.com/p/t0no0g/render-partial-with-collection-has-hidden-index

**Review by ghost — COMMENTED** [review]
cmt

File: app/views/admin/categories/index.html.erb (hunk 3 above)
Comment (SuKay712) [reply app/views/admin/categories/index.html.erb]: kì diệu quá ạ, em cảm ơn anh hehe

**Comment by SuKay712** [comment]
ready

**Pull request closed without merging**